cmd/ci-images-mirror/main.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/sirupsen/logrus"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/rest"
+	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/logrusutil"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+
+	imagev1 "github.com/openshift/api/image/v1"
+
+	quayiociimagesdistributor "github.com/openshift/ci-tools/pkg/controller/quay_io_ci_images_distributor"
+)
+
+var allControllers = sets.New[string](
+	quayiociimagesdistributor.ControllerName,
+)
+
+type options struct {
+	leaderElectionNamespace          string
+	leaderElectionSuffix             string
+	enabledControllers               flagutil.Strings
+	enabledControllersSet            sets.Set[string]
+	dryRun                           bool
+	quayIOCIImagesDistributorOptions quayIOCIImagesDistributorOptions
+}
+
+func (o *options) addDefaults() {
+	o.enabledControllers = flagutil.NewStrings(quayiociimagesdistributor.ControllerName)
+}
+
+type quayIOCIImagesDistributorOptions struct {
+	additionalImageStreamNamespacesRaw flagutil.Strings
+	additionalImageStreamNamespaces    sets.Set[string]
+}
+
+func newOpts() (*options, error) {
+	opts := &options{}
+	opts.addDefaults()
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&opts.leaderElectionNamespace, "leader-election-namespace", "ci", "The namespace to use for leaderelection")
+	fs.StringVar(&opts.leaderElectionSuffix, "leader-election-suffix", "", "Suffix for the leader election lock. Useful for local testing. If set, --dry-run must be set as well")
+	fs.Var(&opts.enabledControllers, "enable-controller", fmt.Sprintf("Enabled controllers. Available controllers are: %v. Can be specified multiple times. Defaults to %v", allControllers.UnsortedList(), opts.enabledControllers.Strings()))
+	fs.BoolVar(&opts.dryRun, "dry-run", true, "Whether to run the controller-manager with dry-run")
```

Comment:
I wouldn't expect `dry-run` to have `true` as a default

Reply:
Done!